### PR TITLE
Fix for #628

### PR DESCRIPTION
**Pull request by rsredsq**
#628

### CODE REVIEW

File: Source/Atomic/Atomic2D/TmxFile2D.cpp [hunk 1]
```diff
@@ -420,25 +416,24 @@ bool TmxFile2D::EndLoad()
             ret = LoadTileSet(childElement);
         else if (name == "layer")
         {
-            TmxTileLayer2D* tileLayer = new TmxTileLayer2D(this);
+            SharedPtr<TmxTileLayer2D> tileLayer (new TmxTileLayer2D(this));
             ret = tileLayer->Load(childElement, info_);
 
-            layers_.Push(tileLayer);
+            layers_.Push(static_cast<SharedPtr<TmxLayer2D>>(tileLayer));
```
Comment (JoshEngebretson) [inline Source/Atomic/Atomic2D/TmxFile2D.cpp]: Looking good, are these static cast's necessary?  It doesn't seem so

**Comment by JoshEngebretson** [comment]
The build failure is a code signing issue on iOS CI box, looking into it, otherwise if the static_cast's aren't necessary let's remove them, thanks!

**Comment by rsredsq** [comment]
hm, yes, probably don't need static_cast, removed

**Comment by JoshEngebretson** [comment]
@AtomicBuildBot Retest this please

**Comment by JoshEngebretson** [comment]
LANDING! :airplane:

**Pull request merged**